### PR TITLE
Auto code show messages

### PR DESCRIPTION
**Pull request by as2388**
Updates REDSS auto-code to handle the four REDSS shows.
 - The show schemes are still TODO until the coding frames have been established
 - ICR is still TODO, but lower priority vs Coda 2 integration because risk is much lower

### CODE REVIEW

File: code_schemes/s01e01.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{
```
Comment (lukechurch) [inline code_schemes/s01e01.json:1]: These should probably be in a different PR, as we should be asking for a review from RDA as well

Comment (lukechurch) [reply code_schemes/s01e01.json:1]: This code scheme looks a bit odd - it's only made out of control codes? Is this because we don't have a coding frame yet?

File: project_redss/lib/dataset_specification.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+from project_redss.redss_code_translators import *
+
+
+class CodingPlan(object):
+    def __init__(self, raw_field, coded_field, coda_name, cleaner=None, code_translator=None,
+                 time_field=None):
+        self.raw_field = raw_field
+        self.coded_field = coded_field
+        self.coda_name = coda_name
```
Comment (lukechurch) [inline project_redss/lib/dataset_specification.py]: Is coda_name the dataset_name? or some other name?

File: project_redss/redss_code_translators.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+from core_data_modules.cleaners import CodeTranslator, Codes
```
Comment (lukechurch) [inline project_redss/redss_code_translators.py:1]: Is it possible to parse these rather than coding them?

File: project_redss/translate_rapid_pro_keys.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import time
+
+from core_data_modules.traced_data import Metadata
+
+
+class TranslateRapidProKeys(object):
+    RAPID_PRO_KEY_MAP = [
```
Comment (lukechurch) [inline project_redss/translate_rapid_pro_keys.py:7]: These map from what to what?

**Review by lukechurch — APPROVED** [review]
Some questions - but looks ok

File: code_schemes/s01e01.json (hunk 1 above)
Comment (as2388) [reply code_schemes/s01e01.json:1]: Exactly. As mentioned in the PR description and the code translator. Only reason it isn't here is because comments break the JSON.

File: project_redss/lib/dataset_specification.py (hunk 2 above)
Comment (as2388) [reply project_redss/lib/dataset_specification.py]: Used to be a few things. Will change now that's only filename

File: project_redss/redss_code_translators.py (hunk 3 above)
Comment (as2388) [reply project_redss/redss_code_translators.py:1]: Yes, but I thought we weren't to rely on/use the display text for anything

File: code_schemes/s01e01.json (hunk 1 above)
Comment (as2388) [reply code_schemes/s01e01.json:1]: Not requesting a research review because there is no scheme to review - just need enough for testing

Comment (lukechurch) [reply code_schemes/s01e01.json:1]: I'm pondering whether we should move to JSON + comments. Something like a subset of JSON5?

If we restrained ourselves to single line comments, without escapes, it's very easy to strip them out before parsing. What do think?

File: project_redss/redss_code_translators.py (hunk 3 above)
Comment (lukechurch) [reply project_redss/redss_code_translators.py:1]: Ah yeah, so if we are going to code them like I'd like to ensure that these codes all exist in the scheme.

The risk I'm concerned about is that someone makes a mistake, regenerates the scheme with new IDs and starts using it. Is there anything in the pipeline to detect that we're attempting to interpret data with the wrong coding scheme?

**Review by lukechurch — APPROVED** [review]

File: project_redss/redss_code_translators.py (hunk 3 above)
Comment (as2388) [reply project_redss/redss_code_translators.py:1]: Still not sure what to do here. Even with a check, maintaining those tables is going to be a pain

Comment (as2388) [reply project_redss/redss_code_translators.py:1]: Will change to put this data in the scheme in a future PR
